### PR TITLE
fix(monorepo): limit engine sync to 'node'

### PR DESCRIPTION
**Pull request by achrinza**
Signed-off-by: Rifa Achrinza <25147899+achrinza@users.noreply.github.com>

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- New tests added or existing tests modified to cover all changes
- Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- API Documentation in code was updated
- Documentation in [/docs/site](../tree/master/docs/site) was updated
- Affected artifact templates in `packages/cli` were updated
- Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by raymondfeng** [comment]
@achrinza Please update https://github.com/strongloop/loopback-next/blob/c21bb77f0efe24a6fecbcf5daab7125b64cce4f7/packages/monorepo/lib/update-package-json.js#L67 to automate the changes.

BTW, do we mandate NPM 7 now?

**Comment by frbuceta** [comment]
> see: #7126
> 
> Signed-off-by: Rifa Achrinza [25147899+achrinza@users.noreply.github.com](mailto:25147899+achrinza@users.noreply.github.com)
> 
> ## Checklist
> * [x]  DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
> * [x]  `npm test` passes on your machine
> * New tests added or existing tests modified to cover all changes
> * Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
> * API Documentation in code was updated
> * Documentation in [/docs/site](../tree/master/docs/site) was updated
> * Affected artifact templates in `packages/cli` were updated
> * [x]  Affected example projects in `examples/*` were updated
> 
> 👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

I wouldn't make this change, we only require npm v7 to loopback development. If we make this changes we also exempt users to use npm v7 in their projects

**Comment by frbuceta** [comment]
> @achrinza Please update
> 
> https://github.com/strongloop/loopback-next/blob/c21bb77f0efe24a6fecbcf5daab7125b64cce4f7/packages/monorepo/lib/update-package-json.js#L67
> 
> to automate the changes.
> BTW, do we mandate NPM 7 now?

Yes, I put it when we changed the lock files both in `loopback-next`(ROOT) and in`@loopback/tsdocs-monorepo`(fixtures/tsdocs-monorepo) so that files are not used again lock v1.

I would not apply this change, we do not have to force users to use NPM >= 7

**Comment by achrinza** [comment]
@frbuceta I noticed #7126 got overwritten by #7287. Is this expected?

**Comment by frbuceta** [comment]
> #7287

I just saw it wasn't Renovate. Look at this 85a45b8a0d6c1216c11256cb511ec6cfbfb4c226

File: packages/monorepo/lib/update-package-json.js [hunk 1]
```diff
@@ -64,7 +64,12 @@ async function updatePackageJsonFiles(options) {
       directory: path.relative(p.rootPath, p.location).replace(/\\/g, '/'),
     };
 
-    pkg.engines = rootPkg.engines;
+    if ((pkg.private === true && Array.isArray(pkg.workspaces)) || pkg.name === '@loopback/tsdocs-monorepo') {
```
Comment (frbuceta) [inline packages/monorepo/lib/update-package-json.js:67]: I have seen that we can do even better. If you change the variable 'isLernaRepo' that is in line 44 to this
```ts
const isLernaRepo = (pkg.private === true && Array.isArray(pkg.workspaces)) || fs.existsSync(path.join(p.location, 'lerna.json')); // rename for example to isMonorepoPackage
```

You can do something like this
```suggestion
    if (isMonorepoPackage) {
```

Comment (frbuceta) [reply packages/monorepo/lib/update-package-json.js:67]: The package @loopback/tsdocs-monorepo goes without saying, it is already included by the lerna package

**Comment by frbuceta** [comment]
@achrinza sorry, I accidentally closed the PR. I have already fixed the version of the lock files. If any of them go back to version v1, Renovate will take care of putting it back in v2

File: packages/monorepo/lib/update-package-json.js (hunk 1 above)
Comment (achrinza) [reply packages/monorepo/lib/update-package-json.js]: Great suggestion; I've applied the changes.

Is there any reason for checking `pkg.private = true`? I think check the existence of `pkg.workspaces` should be suffice.

Comment (frbuceta) [reply packages/monorepo/lib/update-package-json.js]: Normally it is usually put like this but if you prefer to put it that way it is equally fine

**Pull request closed without merging**